stacks/peerdb-server.Dockerfile

```diff
@@ -11,17 +11,27 @@ RUN cargo chef prepare --recipe-path recipe.json
 
 FROM chef AS builder
 ENV OPENSSL_STATIC=1
+ARG CARGO_BUILD_FLAGS="--release"
 RUN apk add --no-cache build-base pkgconfig curl unzip openssl-dev openssl-libs-static
 WORKDIR /root/nexus
 COPY scripts /root/scripts
 RUN /root/scripts/install-protobuf.sh
 COPY --from=planner /root/nexus/recipe.json .
-# Build dependencies - this is the caching Docker layer!
-RUN cargo chef cook --release --recipe-path recipe.json
+# Build dependencies with cache mounts for Cargo registry and target directory
+RUN --mount=type=cache,target=/usr/local/cargo/registry \
+    --mount=type=cache,target=/usr/local/cargo/git \
+    --mount=type=cache,target=/root/nexus/target \
+    cargo chef cook ${CARGO_BUILD_FLAGS} --recipe-path recipe.json
 COPY nexus /root/nexus
 COPY protos /root/protos
 WORKDIR /root/nexus
-RUN cargo build --release --bin peerdb-server
+# Build the actual binary with cache mounts
+RUN --mount=type=cache,target=/usr/local/cargo/registry \
+    --mount=type=cache,target=/usr/local/cargo/git \
+    --mount=type=cache,target=/root/nexus/target \
+    cargo build ${CARGO_BUILD_FLAGS} --bin peerdb-server && \
+    mkdir -p /root/target && \
+    cp target/*/peerdb-server /root/target/
```

Comment:
nit: `target/release/peerdb-server` to be safe?

given that we are mounting everything in target (i.e. `--mount=type=cache,target=/root/nexus/target`), could accidentally pick up non-release builds, and introduce ambiguity.